tests/ceph_ansible/shrink_mon.py

```diff
@@ -37,14 +36,32 @@ def run(ceph_cluster, **kw):
     ceph_installer.node.obtain_root_permissions('/var/log')
     ansible_dir = ceph_installer.ansible_dir
 
-    ceph_installer.exec_command(sudo=True, cmd='cp -R {ansible_dir}/infrastructure-playbooks/shrink-mon.yml '
-                                               '{ansible_dir}/shrink-mon.yml'.format(ansible_dir=ansible_dir))
+    if build.startswith('4'):
+        cmd = 'export ANSIBLE_DEPRECATION_WARNINGS=False ; cd {ansible_dir} ;ansible-playbook -vvvv -e \
+                ireallymeanit=yes infrastructure-playbooks/shrink-mon.yml \
+                -e mon_to_kill={mon_to_kill} -i hosts'.format(ansible_dir=ansible_dir, mon_to_kill=mon_to_kill)
+    else:
+        ceph_installer.exec_command(sudo=True, cmd='cp -R {ansible_dir}/infrastructure-playbooks/shrink-mon.yml '
+                                    '{ansible_dir}/shrink-mon.yml'.format(ansible_dir=ansible_dir))
+        cmd = 'export ANSIBLE_DEPRECATION_WARNINGS=False ; cd {ansible_dir} ;ansible-playbook -vvvv -e \
+                ireallymeanit=yes shrink-mon.yml \
+                -e mon_to_kill={mon_to_kill} -i hosts'.format(ansible_dir=ansible_dir, mon_to_kill=mon_to_kill)
+    out, err = ceph_installer.exec_command(cmd=cmd, long_running=True)
+
+    out1, rc = ceph_installer.exec_command(sudo=True, cmd='cd {ansible_dir};cat hosts'.format(ansible_dir=ansible_dir))
+    inventory = out1.read().decode().split("\n")
+    mons_index = inventory.index("[mons]")
 
-    out, err = ceph_installer.exec_command(cmd='export ANSIBLE_DEPRECATION_WARNINGS=False ; cd {ansible_dir} ; '
-                                               'ansible-playbook -vvvv -e ireallymeanit=yes shrink-mon.yml '
-                                               '-e mon_to_kill={mon_to_kill} -i hosts'.format(ansible_dir=ansible_dir,
-                                                                                              mon_to_kill=mon_to_kill),
-                                           long_running=True)
+    for node in inventory[mons_index:]:
+        if re.match("%s" % matched_short_names[0], node):
+            remove_index = inventory.index(node)
+            break
+    inventory.pop(remove_index)
+    modified_inventory = "\n".join(inventory)
+    hosts_file = ceph_installer.write_file(
+        sudo=True, file_name='{}/hosts'.format(ansible_dir), file_mode='w')
+    hosts_file.write(modified_inventory)
+    hosts_file.flush()
 
```

Comment:
I would suggest print out inventory before and after update, so content of inventory will be good in case of issues/failures.

Reply:
would suggest to create definition to display inventory info.